cpp/system_updater.cc

```diff
@@ -84,16 +123,31 @@ int Main(int argc, char *argv[]) {
     }
 
     std::sort(script_names.begin(), script_names.end(), ScriptLessThan);
+    std::string last_schema, current_schema;
 
     for (const auto &script_name : script_names) {
         LOG_INFO("Running " + script_name);
-        ExecUtil::ExecOrDie(SYSTEM_UPDATES_DIR + "/" + script_name);
+        if (script_name.ends_with(".sh")) {
+            if (dry_run)
+                std::cout << SYSTEM_UPDATES_DIR << "---" << script_name << std::endl;
+            else
+                ExecUtil::ExecOrDie(SYSTEM_UPDATES_DIR + "/" + script_name);
+        }
+        else if (script_name.ends_with(".sql")) {
+            if (dry_run)
+                std::cout << SYSTEM_UPDATES_DIR << " " << script_name << std::endl;
+            else
+                ApplyDatabaseUpdate(&db_connection, SYSTEM_UPDATES_DIR, script_name, &current_schema, last_schema);
+            last_schema = current_schema;
+        }
+        else
+            continue;
```

Comment:
I think this should cause an error including a message, instead of continuing and just skipping it silently.